Automattic/jetpack
# SSO: properly disable "Match by email" by default

#### Changes proposed in this Pull Request:
Jetpack SSO has "match by email" disabled by default in the UI.
This PR brings the code in accordance to that.

The code treats the option `jetpack_sso_match_by_email` as `true` if it does not exist.
That means that even though the UI shows "match by email" disabled by default, it's in fact enabled.
To properly disable it, an admin would need to:
1. Enable "Match by email" to create the option and set it to `true`.
2. Disable "Match by email" to set the option to `false`.

#### Other information:


#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Create a new site and enable SSO. Do not touch the "Match by email" setting, leave it disabled.
2. Create a user with email matching another WPCOM account (let's call them "user 2").
3. Use the "user 2" account to log into WP-admin for the first time via SSO. You should see the "We couldn't find your account" error.
4. Go to "Jetpack -> Settings -> SSO" and enable the "Match by email" setting.
5. Try to log in as "user 2" via SSO again, you should succeed.
6. Go to "Jetpack -> Dashboard" and disconnect "user 2" from WPCOM.
7. Go to "Jetpack -> Settings -> SSO" and disable "Match by email".
8. Log out of WPCOM and try to login via SSO. You should see the "We couldn't find your account" error.
9. Add the following code to the `wp-config.php`: `define( 'WPCC_MATCH_BY_EMAIL', true );`
10. Try to login as "user2" via SSO again. You should succeed.
